### PR TITLE
fix(🐛): Fixes imports and adds RN 0.71 support

### PR DESCRIPTION
**Pull request by emibotti**
## Motivation

* `ScreenContainer`  from `react-native-screen` was imported instead of `SceneContainer` component in `WelcomeScreen` and caused a crash.
* `App.tsx` wasn't found by `index.js` default import (`.. from '.App'`), so it was moved to the root
* Since `react-native-template-typescript` [will no longer be maintained](https://github.com/react-native-community/react-native-template-typescript#this-template-is-deprecated) because of React Native [first-class support for Typescript in version 0.71](https://reactnative.dev/blog/2023/01/03/typescript-first), we decided to don't use this template by default. The only way to use it is by adding a custom `--templateVersion`, to those user that still want to generate an older RN version.
* Added local setup information, for those who want to contribute to the project.

## Screenshot proving that is working

<p align="center">

</p>

### CODE REVIEW

File: src/tools/initCommandHelpers.ts [hunk 1]
```diff
@@ -62,15 +62,23 @@ export const installReactNative = async (
   await handleOperation(projectName, OperationKey.Install, async () => {
     const templateVersion = options['templateVersion']
       ? `@${options['version']}`
-      : ''
-    await execa.command(
-      `npx react-native init ${projectName} --template react-native-template-typescript${templateVersion} --skip-install `
-    )
+      : undefined
```
Comment (jmoliugp) [inline src/tools/initCommandHelpers.ts]: What would you think about this change?
```
const templateVersion = options.templateVersion && `@${options.version}`
```

File: src/tools/initCommandHelpers.ts [hunk 2]
```diff
@@ -62,15 +62,23 @@ export const installReactNative = async (
...
+
+    if (templateVersion) {
+      await execa.command(
+        `npx react-native init ${projectName} --template react-native-template-typescript${templateVersion} --skip-install `
+      )
+    } else {
+      // It will install the latest RN version that already includes first-class TS support
```
Comment (jmoliugp) [inline src/tools/initCommandHelpers.ts:70]: This is one of the few scenarios where I find reasonable to add comments.

[ref](https://github.com/ryanmcdermott/clean-code-javascript#only-comment-things-that-have-business-logic-complexity).

File: src/tools/reactNavigationHelpers.ts [hunk 3]
```diff
@@ -72,6 +72,10 @@ const regenerateAppTsxAndAddBaseScene = async (
     cwd: `${process.cwd()}/${projectName}`,
   })
 
+  await execa.command(`mv src/App.tsx App.tsx`, {
+    cwd: `${process.cwd()}/${projectName}`,
+  })
```
Comment (jmoliugp) [inline src/tools/reactNavigationHelpers.ts:77]: Why did you add this?

**Review by jmoliugp — COMMENTED** [review]
I haven't tested it but is looking good.

Left some minor comments.

File: src/tools/reactNavigationHelpers.ts (hunk 3 above)
Comment (emibotti) [reply src/tools/reactNavigationHelpers.ts:77]: There are some configurations in another file that when creating all the base components and files with all the boilerplate, the `App.tsx` is created in a different folder. That's why here I moved it back to the root folder instead.
We can huddle and take a deeper look to see if we can create the `App.tsx` directly in the root if you want

File: src/tools/initCommandHelpers.ts (hunk 1 above)
Comment (emibotti) [reply src/tools/initCommandHelpers.ts]: It does make sense! Thanks 💪

File: src/tools/reactNavigationHelpers.ts (hunk 3 above)
Comment (jmoliugp) [reply src/tools/reactNavigationHelpers.ts:77]: Makes sense to me 👌🏻

**Review by jmoliugp — APPROVED** [review]
LGTM
